### PR TITLE
Remove jmacd@ from TC; update affiliation

### PR DESCRIPTION
**Pull request by jmacd**
I have resigned from the TC to avoid creating a case of over-representation.
Remove myself from this group. Update my affiliation.

### CODE REVIEW

File: config.yaml
Comment (trask) [inline config.yaml:1]: I'm going to send a separate PR to remove this file (and avoid the confusion) since we're going in a different direction for IaC (and this file is not up-to-date)

File: areas-of-interest.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ concerns.
 - OpenTelemetry Java
 - OpenTelemetry protocol
 
-### [Josh MacDonald](https://github.com/jmacd), Lightstep
+### [Josh MacDonald](https://github.com/jmacd), Microsoft
```
Comment (trask) [inline areas-of-interest.md:44]: ❤️

File: community-members.md [hunk 2]
```diff
@@ -26,7 +26,6 @@ in alphabetical order:
 - [Bogdan Drutu](https://github.com/BogdanDrutu), Snowflake
 - [Carlos Alberto](https://github.com/carlosalberto), Lightstep
 - [Jack Berg](https://github.com/jack-berg), New Relic
-- [Josh MacDonald](https://github.com/jmacd), Lightstep
```
Comment (trask) [inline community-members.md:29]: 😢

**Review by trask — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by mtwo — APPROVED** [review]

File: community-members.md (hunk 2 above)
Comment (svrnm) [reply community-members.md:29]: 😢

**Review by svrnm — APPROVED** [review]

**Review by reyang — APPROVED** [review]

File: community-members.md (hunk 2 above)
Comment (reyang) [reply community-members.md:29]: 😢

**Review by jack-berg — APPROVED** [review]
Adding @jmacd as a spec sponsor in #2447
